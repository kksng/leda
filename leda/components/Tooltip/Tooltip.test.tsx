import React from 'react';
import { render } from '@testing-library/react';
import { I } from '../I';
import { Tooltip } from './index';

jest.mock('./constants', () => ({
  ...jest.requireActual('./constants'),
  defaultArrowSize: 0,
}));

describe('Tooltip SNAPSHOTS', () => {
  it('should render', () => {
    const wrapper = render((
      <Tooltip title="test">test</Tooltip>
    ));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render isOpened state', () => {
    const wrapper = render((
      <Tooltip title="test" isOpen>test</Tooltip>
    ));

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Tooltip ATTRIBUTES', () => {
  it('should accept jsx in title', () => {
    render((
      <Tooltip title={<span style={{ color: 'blue' }}>test</span>} />
    ));

    expect(document.querySelectorAll('span')).toHaveLength(1);
  });

  describe('position prop when open', () => {
    it('should be top by default', () => {
      render((
        <Tooltip isOpen title="test">Test</Tooltip>
      ));

      expect(document.querySelector('.tooltip')).toHaveClass('top');
    });

    it('should be top', () => {
      render((
        <Tooltip isOpen title="test" position="top">test</Tooltip>
      ));

      expect(document.querySelector('.tooltip')).toHaveClass('top');
    });

    it('should be left', () => {
      render((
        <Tooltip isOpen title="test" position="left">test</Tooltip>
      ));

      expect(document.querySelector('.tooltip')).toHaveClass('left');
    });

    it('should be right', () => {
      render((
        <Tooltip isOpen title="test" position="right">test</Tooltip>
      ));

      expect(document.querySelector('.tooltip')).toHaveClass('right');
    });

    it('should be bottom', () => {
      render((
        <Tooltip isOpen title="test" position="bottom">test</Tooltip>
      ));

      expect(document.querySelector('.tooltip')).toHaveClass('bottom');
    });
  });

  it('should have children prop', () => {
    render((
      <Tooltip title="test">
        <div className="level-0" style={{ float: 'right' }}>
          <span className="level-1">test</span>
        </div>
      </Tooltip>
    ));

    expect(document.querySelector('div.level-0')).toBeInTheDocument();

    expect(document.querySelector('div.level-0')).toHaveStyle('float: right');

    expect(document.querySelector('span.level-1')).toBeInTheDocument();

    expect(document.querySelector('span.level-1')).toHaveTextContent('test');
  });

  it('should wrap element', () => {
    render((
      <Tooltip title="test" position="top">
        <I _i-search />
      </Tooltip>
    ));

    expect(document.querySelectorAll('i')).toHaveLength(1);
  });

  it('should wrap elements', () => {
    render((
      <Tooltip title="test" position="top">
        <I _i-search />
        <I _i-search />
      </Tooltip>
    ));

    expect(document.querySelectorAll('i')).toHaveLength(2);
  });
});
