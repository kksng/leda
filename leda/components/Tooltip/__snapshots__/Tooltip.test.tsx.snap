// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip SNAPSHOTS should render 1`] = `
<Tooltip
  title="Tips"
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="tooltip-wrapper"
  >
    Test
  </div>
  <TooltipBody
    position="top"
    style={
      Object {
        "height": "0",
        "left": "-99999px",
        "opacity": "0",
        "whiteSpace": "nowrap",
      }
    }
    theme={
      Object {
        "bottom": "tooltip bottom",
        "left": "tooltip left",
        "right": "tooltip right",
        "top": "tooltip top",
        "wrapper": "tooltip-wrapper",
      }
    }
    title="Tips"
  >
    <Portal
      containerInfo={
        <body>
          <div
            class="tooltip top"
            style="height: 0px; opacity: 0; white-space: nowrap; left: -99999px;"
          >
            <div>
              Tips
            </div>
          </div>
        </body>
      }
    >
      <div
        className="tooltip top"
        style={
          Object {
            "height": "0",
            "left": "-99999px",
            "opacity": "0",
            "whiteSpace": "nowrap",
          }
        }
      >
        <div>
          Tips
        </div>
      </div>
    </Portal>
  </TooltipBody>
</Tooltip>
`;

exports[`Tooltip SNAPSHOTS should render isOpened state 1`] = `
<Tooltip
  isOpen={true}
  title="Tips"
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="tooltip-wrapper"
  >
    Test
  </div>
  <TooltipBody
    position="top"
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "whiteSpace": "nowrap",
      }
    }
    theme={
      Object {
        "bottom": "tooltip bottom",
        "left": "tooltip left",
        "right": "tooltip right",
        "top": "tooltip top",
        "wrapper": "tooltip-wrapper",
      }
    }
    title="Tips"
  >
    <Portal
      containerInfo={
        <body>
          <div
            class="tooltip top"
            style="height: auto; opacity: 1; white-space: nowrap;"
          >
            <div>
              Tips
            </div>
          </div>
        </body>
      }
    >
      <div
        className="tooltip top"
        style={
          Object {
            "height": "auto",
            "opacity": 1,
            "whiteSpace": "nowrap",
          }
        }
      >
        <div>
          Tips
        </div>
      </div>
    </Portal>
  </TooltipBody>
</Tooltip>
`;
